File: ios/PIP/SettingsModule.m
#import <Foundation/Foundation.h>

#if __has_include(<React/RCTBridgeModule.h>)
#import <React/RCTBridgeModule.h>
#elif __has_include("RCTBridgeModule.h")
#import "RCTBridgeModule.h"
#else
#import "React/RCTBridgeModule.h"
#endif

@interface RCT_EXTERN_MODULE(SettingsModule, NSObject)

RCT_EXTERN_METHOD(getVolume:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject)
RCT_EXTERN_METHOD(getBrightness:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject)
RCT_EXTERN_METHOD(setVolume:(float)volume)
RCT_EXTERN_METHOD(setBrightness:(CGFloat)brightness)

@end

File: ios/PIP/SettingsModule.swift
import Foundation
import UIKit
import MediaPlayer

@objc(SettingsModule)
class SettingsModule: NSObject {
  
  @objc
  static func requiresMainQueueSetup() -> Bool {
    return true
  }
  
  @objc
  func getVolume(_ resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
    DispatchQueue.main.async {
      let volume = AVAudioSession.sharedInstance().outputVolume
      resolve(volume)
    }
  }
  
  @objc
  func getBrightness(_ resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
    DispatchQueue.main.async {
      let brightness = UIScreen.main.brightness
      resolve(brightness)
    }
  }
  
  @objc
  func setVolume(_ volume: Float) {
    // Note: iOS restricts direct system volume changes for privacy
    // This is a common way to suggest a volume change or use MPVolumeView
    print("Setting volume on iOS to: \(volume)")
  }
  
  @objc
  func setBrightness(_ brightness: CGFloat) {
    DispatchQueue.main.async {
      UIScreen.main.brightness = brightness
    }
  }
}
